File: 237-DeleteNodeInALinkedList/237-DeleteNodeInALinkedList.java
// Last updated: 9/14/2025, 9:15:16 AM
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        ListNode nextNode = node.next;
        node.val = nextNode.val;
        node.next = nextNode.next;
    }
}

File: 237-DeleteNodeInALinkedList/237-DeleteNodeInALinkedList.c
// Last updated: 9/14/2025, 9:15:19 AM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
void deleteNode(struct ListNode* node) {
    node->val=node->next->val;
    node->next=node->next->next;
}